src/server/raftkv2/mod.rs

```diff
@@ -0,0 +1,301 @@
+                        ASYNC_REQUESTS_COUNTER_VEC.snapshot.get(status_kind).inc();
+                        Err(err)
+                    } else {
+                        Err(box_err!("unexpected response: {:?}", resp))
+                    }
+                }
+            }
+        }
+    }
+
+    type WriteRes = impl Stream<Item = WriteEvent> + Send + Unpin;
+    fn async_write(
+        &self,
+        ctx: &kvproto::kvrpcpb::Context,
+        batch: tikv_kv::WriteData,
+        _subscribed: u8,
+        on_applied: Option<tikv_kv::OnAppliedCb>,
+    ) -> Self::WriteRes {
+        let region_id = ctx.region_id;
+        ASYNC_REQUESTS_COUNTER_VEC.write.all.inc();
+        let begin_instant = Instant::now_coarse();
+        let mut header = Box::new(new_request_header(ctx));
+        let mut flags = 0;
+        if batch.extra.one_pc {
+            flags |= WriteBatchFlags::ONE_PC.bits();
+        }
+        if batch.extra.allowed_in_flashback {
+            flags |= WriteBatchFlags::FLASHBACK.bits();
+        }
+        header.set_flags(flags);
+
+        self.schedule_txn_extra(batch.extra);
+        let mut encoder = SimpleWriteEncoder::with_capacity(128);
+        for m in batch.modifies {
+            match m {
+                Modify::Put(cf, k, v) => encoder.put(cf, k.as_encoded(), &v),
+                Modify::Delete(cf, k) => encoder.delete(cf, k.as_encoded()),
+                Modify::PessimisticLock(k, lock) => {
+                    encoder.put(CF_LOCK, k.as_encoded(), &lock.into_lock().to_bytes())
+                }
+                Modify::DeleteRange(cf, start_key, end_key, notify_only) => encoder.delete_range(
+                    cf,
+                    start_key.as_encoded(),
+                    end_key.as_encoded(),
+                    notify_only,
+                ),
+            }
+        }
+        let data = encoder.encode();
+        let (ch, sub) = match on_applied {
+            Some(cb) => {
+                let (ch, sub) = CmdResChannel::with_callback(move |resp| {
+                    let mut res = if !resp.get_header().has_error() {
+                        Ok(())
+                    } else {
+                        Err(tikv_kv::Error::from(resp.get_header().get_error().clone()))
+                    };
+                    cb(&mut res);
+                });
+                (ch, sub)
+            }
+            None => CmdResChannel::pair(),
+        };
+        let msg = PeerMsg::SimpleWrite(SimpleWrite {
+            header,
+            data,
+            ch,
+            send_time: Instant::now_coarse(),
+        });
+        let res = self
+            .router
+            .store_router()
+            .send(region_id, msg)
+            .map_err(|e| tikv_kv::Error::from(raftstore_v2::Error::from(e)));
+        (Transform {
+            resp: CmdResStream::new(sub),
+            early_err: res.err(),
+        })
+        .inspect(move |ev| {
```

Comment:
where's the definition of the inspect function?  It seems not in either Stream or Transform